mail_factory/contrib/auth/forms.py

```diff
@@ -1,54 +1,51 @@
 # -*- coding: utf-8 -*-
-from django.conf import settings
-from django.contrib.auth.forms import PasswordResetForm
-from django.contrib.sites.models import get_current_site
+
+from django.contrib.auth.forms import \
+    PasswordResetForm as DjangoPasswordResetForm
 from django.contrib.auth.tokens import default_token_generator
-from django.utils.http import int_to_base36
+from django.contrib.sites.shortcuts import get_current_site
+from django.utils.encoding import force_bytes, force_text
+from django.utils.http import urlsafe_base64_encode
 
-from .mails import PasswordResetMail
 from mail_factory import factory
 
+from .mails import PasswordResetMail
+
 
-class PasswordResetForm(PasswordResetForm):
+class PasswordResetForm(DjangoPasswordResetForm):
     """MailFactory PasswordReset alternative."""
 
-    def save(self, domain_override=None,
-             subject_template_name=None,  # Not used anymore
-             email_template_name=None,  # Mail Factory template name
-             mail_object=None,  # Mail Factory Mail object
-             use_https=False, token_generator=default_token_generator,
-             from_email=None, request=None):
+    def mail_factory_email(
+            self, domain_override=None, email_template_name=None,
+            use_https=False, token_generator=default_token_generator,
+            from_email=None, request=None, extra_email_context=None):
         """
         Generates a one-use only link for resetting password and sends to the
         user.
         """
-        for user in self.users_cache:
+        email = self.cleaned_data["email"]
+        for user in self.get_users(email):
             if not domain_override:
                 current_site = get_current_site(request)
                 site_name = current_site.name
                 domain = current_site.domain
             else:
                 site_name = domain = domain_override
-
-            context_params = {
-                'email': user.email,
+            context = {
+                'email': email,
                 'domain': domain,
                 'site_name': site_name,
-                'uid': int_to_base36(user.pk),
+                'uid': force_text(urlsafe_base64_encode(force_bytes(user.pk))),
```

Comment:
Don't we have the risk to generate not retro-compatible reset password mails?

Reply:
Hum... good question... It was validated in the previous PR so I did pay attention to this.

Reply:
Indeed.
This same modification was made in django in 2013 to allow to work with non-interger user primary keys.
At the time, they kept a backward compatible url and view, which was removed in django 1.7.

Reply:
Thx :smile:

Reply:
AFAICT this doesn't impact our production apps. It could impact someone using django-mail-factory to send reset passwords, but I think we can take that risk.